core/src/peer_id.rs

```diff
@@ -35,7 +35,15 @@ pub struct PeerId {
 
 impl fmt::Debug for PeerId {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "PeerId({})", self.to_base58())
+        f.debug_tuple("PeerId")
+            .field(&self.to_base58())
+            .finish()
+    }
+}
+
+impl fmt::Display for PeerId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.to_base58(), f)
```

Comment:
```suggestion
        self.to_base58().fmt(f)
```

Reply:
Why is that better?

Reply:
It looks less noisy to me and is shorter.

Reply:
But if we imported `Debug` and `Display` directly instead of through `fmt`, we'd get a compilation error.

Reply:
Sure, but we did not.